### PR TITLE
Solves RegEx issue.

### PR DESCRIPTION
**Pull request by xCh12i5**
**By submitting this pull request, I confirm the following:** 
*please fill any appropriate checkboxes, e.g: [X]*

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md), as well as this entire template.
- [X] I have made only one major change in my proposed changes.
- [X] I have commented my proposed changes within the code.
- [ ] I have tested my proposed changes, and have included unit tests where possible.
- [X] I am willing to help maintain this change if there are issues with it later.
- [X] I give this submission freely and claim no ownership.
- [X] It is compatible with the [EUPL 1.2 license](https://opensource.org/licenses/EUPL-1.1)
- [X] I have squashed any insignificant commits. ([`git rebase`](http://gitready.com/advanced/2009/02/10/squashing-commits-with-rebase.html))

Please make sure you [Sign Off](https://github.com/pi-hole/pi-hole/wiki/How-to-signoff-your-commits.) all commits. Pi-hole enforces the [DCO](https://github.com/pi-hole/pi-hole/wiki/Contributing-to-the-project).

---
**What does this PR aim to accomplish?:**
[Issue 2463](https://github.com/pi-hole/pi-hole/issues/2463)

**How does this PR accomplish the above?:**
Replaced the original RegEx with the tested RegEx proposed in issue 2463. IP-addresses should not be matched anymore.

**What documentation changes (if any) are needed to support this PR?:**
n/a

---
* You must follow the template instructions. Failure to do so will result in your pull request being closed.
* Please respect that Pi-hole is developed by volunteers, who can only reply in their spare time.

### CODE REVIEW

File: gravity.sh [hunk 1]
```diff
@@ -338,11 +338,11 @@ gravity_ParseFileIntoDomains() {
     # Awk -F splits on given IFS, we grab the right hand side (chops trailing #coments and /'s to grab the domain only.
```
[the remark this answers is not in the document]
Comment (xCh12i5) [reply gravity.sh]: Done.

File: gravity.sh [hunk 2]
```diff
@@ -335,14 +335,16 @@ gravity_ParseFileIntoDomains() {
...
-    tr '[:upper:]' '[:lower:]' | \
-    sed -r '/(\/|#).*$/d' | \
-    sed -r 's/^.*\s+//g' | \
-    sed -r '/([^\.]+\.)+[^\.]{2,}/!d' >  ${destination}
+    # 1) Remove carriage returns
+    # 2) Convert all characters to lowercase
+    # 3) Remove lines containing "#" or "/"
+    # 4) Remove leading tabs, spaces, etc.
+    # 5) Delete lines not matching domain names
+    < ${source} tr -d "\r" | \
```
Comment (dschaper) [inline gravity.sh]: Stickler isn't too great on what it's complaining about. The original issue was the variable expansion:

`< "${source}" tr -d \r | \`

**Comment by pralor** [comment]
This pull request has been mentioned on **Pi-hole Userspace**. There might be relevant details there:

https://discourse.pi-hole.net/t/strip-ip-address-from-blocklist/17665/3

**Comment by DL6ER** [comment]
@dschaper I'm going to merge this if you have no objections.

**Comment by dschaper** [comment]
Yeah, I have no objections, I didn't approve or merge it since I added commits to the PR and I didn't want to merge my own work.

**Comment by dschaper** [comment]
@PromoFaux Does this affect the regex changes we just merged in to release v4.3.2?

**Pull request merged**